Signal/src/ViewControllers/ConversationView/Cells/OWSSystemMessageCell.m

```diff
@@ -266,7 +271,11 @@ - (CGSize)cellSizeForViewWidth:(int)viewWidth contentWidth:(int)contentWidth
     result.height += self.topVMargin;
     result.height += self.bottomVMargin;
 
-    [self applyTitleForInteraction:interaction label:self.titleLabel];
+    // FIXME pass in transaction from the uiDBConnection.
```

Comment:
Are you going to do this for the same release (v2.25?)?

Reply:
If time allows it would be nice, but likely not for this release. To be clear, I don't think this a regression - we were previously creating a transaction here too, it's just that it was buried deep down in `applyTitleForInteraction:label` -> `description`.

`description` opened a transaction from the shared model.dBConnection not the uiDBConnection.

Instead we're now using `previewTextWithTransaction:` instead of `description` for any user facing text.

But yes! I'd like to do a similar audit of the conversation view and apply the same patterns. The existing ConversationViewItem (which maybe should be renamed to `InteractionViewModel` or `ConversationItemViewModel`) gets us part of the way there, but it still relies on self-created (non uiDBConnection transactions).

Though, I think it'll be a bigger endeavor, since the implicit-transaction pattern is used not just in the ConversationViewController, but also in the ConversationLayout classes (and maybe some other places).